lib/connection/connection_config.js

```diff
@@ -11,42 +11,74 @@ const NativeTypes = require('./result/data_types').NativeTypes;
 const GlobalConfig = require('../global_config');
 const authenticationTypes = require('../authentication/authentication').authenticationTypes;
 
+const PARAMS =
+{
+  ACCOUNT: 'ACCOUNT',
+  REGION: 'REGION',
+  HOST: 'HOST',
+  ACCESS_URL: 'ACCESSURL',
+  USERNAME: 'USERNAME',
+  PASSWORD: 'PASSWORD',
+  AUTHENTICATOR: 'AUTHENTICATOR',
+  PROXY_HOST: 'PROXYHOST',
+  PROXY_PORT: 'PROXYPORT',
+  SERVICE_NAME: 'SERVICENAME',
+  PRIVATE_KEY: 'PRIVATEKEY',
+  PRIVATE_KEY_PATH: 'PRIVATEKEYPATH',
+  PRIVATE_KEY_PASS: 'PRIVATEKEYPASS',
+  TOKEN: 'TOKEN',
+  WAREHOUSE: 'WAREHOUSE',
+  DATABASE: 'DATABASE',
+  SCHEMA: 'SCHEMA',
+  ROLE: 'ROLE',
+  STREAM_RESULT: 'STREAMRESULT',
+  FETCH_AS_STRING: 'FETCHASSTRING',
+  CLIENT_SESSION_KEEP_ALIVE: 'CLIENTSESSIONKEEPALIVE',
+  CLIENT_SESSION_KEEP_ALIVE_HEARTBEAT_FREQUENCY: 'CLIENTSESSIONKEEPALIVEHEARTBEATFREQUENCY',
+  JS_TREAT_INTEGER_AS_BIGINT: 'JSTREATINTEGERASBIGINT',
+  SESSION_TOKEN: 'SESSIONTOKEN',
+  MASTER_TOKEN: 'MASTERTOKEN',
+  SESSION_TOKEN_EXPIRATION_TIME: 'SESSIONTOKENEXPIRATIONTIME',
+  MASTER_TOKEN_EXPIRATION_TIME: 'MASTERTOKENEXPIRATIONTIME',
+  AGENT_CLASS: 'AGENTCLASS'
+};
+
```

Comment:
Can you change these to be snake case? Its a little hard to read as its currently written

Reply:
Changing the values to snake case would require the connection string parameters to be typed in snake case as well. One word parameters will be unaffected but parameters with two or more will be affected
```
var connection = snowflake.createConnection({
  access_url: "test",
  private_key: "test",
  proxy_host: "test",
  account: "test",
  username: "test",
  password: "test",
```

It's possible to keep the connection string the same using a module to convert to snake case but it won't be case insensitive
```
var connection = snowflake.createConnection({
  accessUrl: "test",
  aCcount: "test",
```
In this case, "accessUrl" is correctly converted to "access_url" but "aCcount" would be converted to "a_C_count"

The parameters were converted to uppercase so it would be case insensitive but it does become hard to read. It could be converted to lowercase if it's easier